app/src/main/java/org/oppia/app/player/state/itemviewmodel/DragAndDropSortInteractionViewModel.kt

```diff
@@ -50,14 +60,65 @@ class DragAndDropSortInteractionViewModel(
     return htmlItems.joinToString(separator = "<br>")
   }
 
+  /** Returns whether the grouping is allowed or not for [DragAndDropSortInteractionViewModel]. */
+  fun getGroupingStatus(): Boolean {
+    return allowMultipleItemsInSamePosition
+  }
+
+  fun updateList(
+    itemIndex: Int,
+    adapter: RecyclerView.Adapter<RecyclerView.ViewHolder>
+  ) {
+    val item = choiceItems[itemIndex]
+    val nextItem = choiceItems[itemIndex + 1]
+    nextItem.htmlContent = StringList.newBuilder().addAllHtml(nextItem.htmlContent.htmlList)
+      .addAllHtml(item.htmlContent.htmlList)
+      .build()
+    choiceItems[itemIndex + 1] = nextItem
+
+    choiceItems.removeAt(itemIndex)
+
+    choiceItems.forEachIndexed { index, dragDropInteractionContentViewModel ->
+      dragDropInteractionContentViewModel.itemIndex = index
+      dragDropInteractionContentViewModel.listSize = choiceItems.size
+    }
+    //to update the content of grouped item
+    (adapter as BindableAdapter<*>).setDataUnchecked(choiceItems)
+  }
+
+  fun unlinkElement(itemIndex: Int, adapter: RecyclerView.Adapter<RecyclerView.ViewHolder>) {
+    val item = choiceItems[itemIndex]
+    choiceItems.removeAt(itemIndex)
+    item.htmlContent.htmlList.asReversed().forEach {
+      choiceItems.add(0, DragDropInteractionContentViewModel(StringList.newBuilder()
+        .addHtml(it)
+        .build()
+      ,0
```

Comment:
Please format this & other code conventionally to how we format other code in the codebase.

Reply:
done